pubsub/subscriptions.js

```diff
@@ -13,149 +13,128 @@
+  // Get the subscription if it exists, otherwise create the subscription
+  // See https://googlecloudplatform.github.io/google-cloud-node/#/docs/pubsub/latest/pubsub/topic?method=subscribe
+  topic.subscribe(subscriptionName, function (err, subscription, apiResponse) {
     if (err) {
       return callback(err);
     }
 
     console.log('Created subscription %s to topic %s', subscriptionName, topicName);
-    return callback(null, subscription);
+    return callback(null, subscription, apiResponse);
   });
 }
-// [END create_subscription]
-
-// [START delete_subscription]
-/**
- * Delete a subscription.
- *
- * @param {string} subscriptionName Name of the subscription to delete.
- * @param {Function} callback The callback function.
- */
-function deleteSubscription (subscriptionName, callback) {
-  if (!subscriptionName) {
-    return callback(new Error('"subscriptionName" is required!'));
-  }
 
+function deleteSubscription (subscriptionName, callback) {
+  var pubsub = PubSub();
   var subscription = pubsub.subscription(subscriptionName);
 
   // Delete the subscription
-  subscription.delete(function (err) {
+  // See https://googlecloudplatform.github.io/google-cloud-node/#/docs/pubsub/latest/pubsub/subscription?method=delete
+  subscription.delete(function (err, apiResponse) {
     if (err) {
       return callback(err);
     }
 
     console.log('Deleted subscription: %s', subscriptionName);
-    return callback(null);
+    return callback(null, apiResponse);
+  });
+}
+
+function getSubscriptionMetadata (subscriptionName, callback) {
+  var pubsub = PubSub();
+  var subscription = pubsub.subscription(subscriptionName);
+
+  // Get the metadata for the specified subscription
+  // See https://googlecloudplatform.github.io/google-cloud-node/#/docs/pubsub/latest/pubsub/subscription?method=getMetadata
+  subscription.getMetadata(function (err, metadata) {
+    if (err) {
+      return callback(err);
+    }
+
+    console.log('Got metadata for subscription: %s', subscriptionName);
+    return callback(null, metadata);
   });
 }
-// [END delete_subscription]
-
-// [START get_all_subscriptions]
-/**
- * List all subscriptions for the specified topic, or list all subcriptions for
- * all topics.
- *
- * @param {string} [topicName] The name of the topic. If omitted, list all
- * subscriptions.
- * @param {Function} callback The callback function.
- */
+
 function listSubscriptions (topicName, callback) {
-  var options = {};
-  if (typeof topicName === 'string') {
-    // Optionally find subscriptions for a specific topic
-    options.topic = pubsub.topic(topicName);
-  }
-  pubsub.getSubscriptions(options, function (err, subscriptions) {
+  var pubsub = PubSub();
+  var topic = pubsub.topic(topicName);
+
+  // List all subscriptions for the specified topic
+  // See https://googlecloudplatform.github.io/google-cloud-node/#/docs/pubsub/latest/pubsub/topic?method=getSubscriptions
+  topic.getSubscriptions(function (err, subscriptions) {
```

Comment:
I've been leaving these off because we're taking advantage of the auto-pagination